foundations/src/telemetry/tokio_runtime_metrics/mod.rs

```diff
@@ -0,0 +1,119 @@
+//! | tokio_runtime_worker_local_schedules_total       | [`tokio::runtime::RuntimeMetrics::worker_local_schedule_count`]     | runtime_name?, runtime_id?, worker_idx |
+//! | tokio_runtime_worker_overflows_total             | [`tokio::runtime::RuntimeMetrics::worker_overflow_count`]           | runtime_name?, runtime_id?, worker_idx |
+//! | tokio_runtime_worker_local_queue_depth           | [`tokio::runtime::RuntimeMetrics::worker_local_queue_depth`]        | runtime_name?, runtime_id?, worker_idx |
+//! | tokio_runtime_worker_mean_poll_time_micros       | [`tokio::runtime::RuntimeMetrics::worker_mean_poll_time`]           | runtime_name?, runtime_id?, worker_idx |
+//!
+//! # Example
+//! ```no_run
+//! # use std::thread;
+//! # use std::time::Duration;
+//! # use foundations::telemetry::tokio_runtime_metrics::{record_runtime_metrics_sample, register_runtime_with_monitor};
+//! // create and monitor 8 runtimes spawned on other threads, then poll in the background from this one
+//! for i in 0..8 {
+//!     let r = tokio::runtime::Builder::new_current_thread().enable_all().build().unwrap();
+//!     
+//!     register_runtime_with_monitor(None, Some(i), r.handle());
+//!
+//!     thread::spawn(move || r.block_on(async {
+//!         loop {
+//!             println!("Poll");
+//!             tokio::time::sleep(Duration::from_secs(1)).await;
+//!         }
+//!     }));
+//! }
+//!
+//! loop {
+//!     record_runtime_metrics_sample();
+//!
+//!     // record metrics roughly twice a second
+//!     std::thread::sleep(Duration::from_secs_f32(0.5));
+//! }
+//! ```
+//!
+//! [`foundations metrics api`]: crate::telemetry::metrics
+
+use crate::telemetry::tokio_runtime_metrics::runtime_handle::RuntimeHandle;
+use parking_lot::Mutex;
+use std::sync::Arc;
+use tokio::runtime::Handle;
+
+mod metrics;
+mod runtime_handle;
+
+static MONITOR: Mutex<RuntimeMonitor> = Mutex::new(RuntimeMonitor::new());
+
+/// Add a runtime to the global monitor, optionally with a name and/or id in case you are monitoring multiple runtimes.
+///
+/// Runtimes should be uniquely identifiable by label/id tuple.
+pub fn register_runtime_with_monitor(
+    runtime_name: Option<Arc<str>>,
+    runtime_id: Option<usize>,
+    handle: &Handle,
+) {
+    let mut monitor = MONITOR.lock();
+
+    assert!(
+        monitor
+            .runtimes
+            .iter()
+            .all(|handle| (handle.runtime_name.as_ref(), handle.runtime_id)
+                != (runtime_name.as_ref(), runtime_id)),
+        "Runtime name and index tuples must be unique"
+    );
+
+    monitor.runtimes.push(RuntimeHandle {
+        runtime_name,
+        runtime_id,
+        handle: handle.clone(),
+    });
+}
+
+/// Record a sample of runtime metrics for each tracked runtime.
+pub fn record_runtime_metrics_sample() {
+    MONITOR
+        .lock()
+        .runtimes
+        .iter_mut()
+        .for_each(RuntimeHandle::record_sample)
+}
+
+#[derive(Default)]
+struct RuntimeMonitor {
```

Comment:
Now when all API is exposed as functions that work with statics we can remove this struct and just store a static vector. At the same time `RuntimeHandle` is ambiguous name, so probably let's rename `RuntimeHandle` to `RuntimeMonitor`

Reply:
I left it this way since this seems slightly cleaner, but I don't really care too much either way here so I'll make this change.